### PR TITLE
[WFLY-19370] Changed order of subsystem cleanup

### PR DESCRIPTION
**Pull request by rsearls**
https://issues.redhat.com/browse/WFLY-19370

### CODE REVIEW

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/jwt/propagation/JWTIdentityPropagationTestCase.java [hunk 1]
```diff
@@ -372,16 +372,17 @@ public void setup(ManagementClient managementClient, String containerId) throws
         @Override
         public void tearDown(final ManagementClient managementClient, final String containerId) throws Exception {
             try (CLIWrapper cli = new CLIWrapper(true)) {
-                cli.sendLine(String.format("/subsystem=ejb3/application-security-domain=%s:remove()", EJB_SECURITY_DOMAIN));
-                cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:remove()", EJB_SECURITY_DOMAIN));
-                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:remove()", EJB_SECURITY_REALM));
-                cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:remove()", ANOTHER_EJB_SECURITY_DOMAIN));
-                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:remove()", ANOTHER_EJB_SECURITY_REALM));
                 cli.sendLine(String.format("/subsystem=elytron/virtual-security-domain=%s:remove()", SINGLE_DEPLOYMENT + ".ear"));
```
Comment (bstansberry) [inline testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/jwt/propagation/JWTIdentityPropagationTestCase.java:375]: Setup does this:

```
cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:write-attribute(name=trusted-virtual-security-domains, value=[%s, %s])",
                        EJB_SECURITY_DOMAIN, SINGLE_DEPLOYMENT + ".ear", EAR_DEPLOYMENT_WITH_MP_JWT + ".ear")); 
```

That means EJB_SECURITY_DOMAIN has a reference to those two virtual-security-domain resources, which will prevent them being removed.

So I think this is needed before SINGLE_DEPLOYMENT or EAR_DEPLOYMENT_WITH_MP_JWT can be removed:

```
cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:undefine-attribute(name=trusted-virtual-security-domains)", EJB_SECURITY_DOMAIN)); 
```

Comment (rsearls) [reply testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/jwt/propagation/JWTIdentityPropagationTestCase.java:377]: done

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @rsearls. This is a good improvement, but I'm going to leave the JIRA unresolved for now as I don't expect correcting these tearDown flaws will affect the behavior of an individual @Test method. The tearDown executes after all the test methods are done (similar to an @AfterClass method).

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/jwt/propagation/JWTIdentityPropagationTestCase.java (hunk 1 above)
Comment (rsearls) [reply testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/jwt/propagation/JWTIdentityPropagationTestCase.java:377]: In the current push.  I have fixed a cli failure during tearDown.
The changes are primarily enabling logging of elytron and apache.  I added more detail to the failing assertion.  These changes will be removed in the future.

**Pull request merged**